libr/io/io_map.c

```diff
@@ -606,17 +613,20 @@ R_API bool r_io_map_write_to_overlay(RIOMap *map, ut64 addr, const ut8 *buf, int
 			chunk = node? (MapOverlayChunk *)node->data: NULL;
 		}
 		if (chunk && r_itv_end (search_itv) >= r_itv_begin (chunk->itv)) {
-			ut8 *ptr = realloc (chunk->buf,
-				(r_itv_end (chunk->itv) - r_itv_begin (search_itv)) * sizeof (ut8));
-			if (!ptr) {
-				return false;
+			ut64 new_size = r_itv_end (chunk->itv) - r_itv_begin (search_itv);
+			st64 new_size2 = new_size * sizeof (ut8);
+			if (new_size > 0) {
```

Comment:
always true, because of check in line 615

Reply:
cant be zero? because in the sample object we tested yesterday it was donig some realloc(0) which is UB

Reply:
that got fixed